spec/models/post_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it 'should validate presence of title' do
+    post = Post.new(title: '')
+    expect(post.valid?).to be false
+    expect(post.errors[:title]).to include("can't be blank")
+  end
+
+  it 'should validate length of title to be at most 250 characters' do
+    post = Post.new(title: 'a' * 251)
+    expect(post.valid?).to be false
+    expect(post.errors[:title]).to include('is too long (maximum is 250 characters)')
+  end
+
+  it 'should validate comments_counter to be an integer greater than or equal to 0' do
+    post = Post.new(comments_counter: -1)
+    expect(post.valid?).to be false
+    expect(post.errors[:comments_counter]).to include('must be greater than or equal to 0')
+  end
+
+  it 'should validate likes_counter to be an integer greater than or equal to 0' do
+    post = Post.new(likes_counter: -1)
+    expect(post.valid?).to be false
+    expect(post.errors[:likes_counter]).to include('must be greater than or equal to 0')
+  end
+
+  describe 'Methods' do
+    describe '#update_user_posts_counter' do
+      it 'increases the authors posts_counter by 1' do
+        user = User.new(posts_counter: 0)
+        user.save
+        post = Post.new(author: user)
+
+        starting_posts_counter = user.posts_counter
+        post.update_user_posts_counter
+        ending_posts_counter = user.posts_counter
+        expect(ending_posts_counter - starting_posts_counter).to eq(0)
```

Comment:
- [ ] Good job working on the project so far 👏🏽 , you have done a fantastic job! However, some tests still fail as the previous reviewer pointed out. This is partly because you used `comments_counter` in your models and spec files, but you have it as `comment_counter` in the DB. Kindly make the changes, confirm that all the tests pass and resubmit for a review 👍🏽 

<img width="1115" alt="Screenshot 2024-01-13 at 17 37 47" src="https://github.com/yusufyusie/blog_app/assets/64774936/308f6031-5255-4c5c-9f01-d11624f67236">

Reply:
well resolved! ✔️